### PR TITLE
[RHDEVDOCS-3107] [OP] Document integration of Tekton Hub with pipeline builder

### PR DESCRIPTION
**Pull request by rishumehra**
**Purpose**:

- Using the updated UI of Pipeline builder in the developer console
- How to add tasks in a pipeline using **Add task** and **Quick Search**

**PR information**:

- Aligned team: DevConsole
- For branches: OpenShift 4.9
- JIRA: https://issues.redhat.com/browse/RHDEVDOCS-3107
- Direct link to doc preview: https://deploy-preview-36625--osdocs.netlify.app/openshift-enterprise/latest/cicd/pipelines/working-with-pipelines-using-the-developer-perspective?utm_source=github&utm_campaign=bot_dp#op-constructing-pipelines-using-pipeline-builder_working-with-pipelines-using-the-developer-perspective
- SME and QE Review: @karthikjeeyar 
- Peer Review: @sounix000 @bburt-rh

### CODE REVIEW

File: modules/op-constructing-pipelines-using-pipeline-builder.adoc [hunk 1]
```diff
@@ -30,13 +30,20 @@ image::op-pipeline-yaml.png[]
...
 
 .. In the *Name* field, enter a unique name for the pipeline.
-.. In the *Tasks* section, select a task from the *Select Task* list to add a task to the pipeline. In this example, use the *s2i-nodejs* task.
-*** To add sequential tasks to the pipeline, click the plus icon to the right or left of the task, and from the *Select Task* list, select the task you want to add to the pipeline. For this example, use the plus icon to the right of the *s2i-nodejs* task to add an *openshift-client* task.
-*** To add a parallel task to the existing task, click the plus icon displayed next to the task, and from the *Select Task* list, select the parallel task you want to add to the pipeline.
+.. In the *Tasks* section, click *Add task*, search for a task in the quick search, select the required task from the list and click *Add* or *Install and add*. In this example, use the *s2i-nodejs* task.
++
+[NOTE]
+====
+The search list contains all the Tekton Hub tasks and tasks available on the cluster. Also, if a task is already installed it will show *Add* to add the task whereas it will show *Install and add* to install and add the task.
```
Comment (karthikjeeyar) [inline modules/op-constructing-pipelines-using-pipeline-builder.adoc:37]: Should we also mention the upgrade path as well in the note ?

Update Path:
If a specific version of a task (eg: 0.1) is already installed on the cluster and when the user chooses to update the task version to 0.2, then "Update and add" option will be available in the UI with a warning note as below.

Screenshot 2021-09-22 at 9 44 15 AM

**Review by karthikjeeyar — COMMENTED** [review]
One more thing worth mentioning is the modal that appears on clicking Add task button, is **draggable and resizable**. Users can move around to see the task underneath the modal and also they could use this resize feature to see more content.

File: modules/op-constructing-pipelines-using-pipeline-builder.adoc (hunk 1 above)
Comment (rishumehra) [reply modules/op-constructing-pipelines-using-pipeline-builder.adoc:37]: Thanks for your input @karthikjeeyar I'll try to repurpose the content with the suggestion.

**Comment by rishumehra** [comment]
> One more thing worth mentioning is the modal that appears on clicking Add task button, is **draggable and resizable**. Users can move around to see the task underneath the modal and also they could use this resize feature to see more content.

Dragging and resizing the quick search won't add much value to the content so it's not necessary to mention the same.

File: modules/op-constructing-pipelines-using-pipeline-builder.adoc [hunk 2]
```diff
@@ -30,13 +30,20 @@ image::op-pipeline-yaml.png[]
 Configure your pipeline using the *Pipeline Builder*:
 
 .. In the *Name* field, enter a unique name for the pipeline.
-.. In the *Tasks* section, select a task from the *Select Task* list to add a task to the pipeline. In this example, use the *s2i-nodejs* task.
-*** To add sequential tasks to the pipeline, click the plus icon to the right or left of the task, and from the *Select Task* list, select the task you want to add to the pipeline. For this example, use the plus icon to the right of the *s2i-nodejs* task to add an *openshift-client* task.
-*** To add a parallel task to the existing task, click the plus icon displayed next to the task, and from the *Select Task* list, select the parallel task you want to add to the pipeline.
+.. In the *Tasks* section, click *Add task*, search for a task in the quick search, select the required task from the list and click *Add* or *Install and add*. In this example, use the *s2i-nodejs* task.
```
Comment (sounix000) [inline modules/op-constructing-pipelines-using-pipeline-builder.adoc:33]: The first sentence seems to be quite complex, in the sense that it contains several sequential actions. Can we break it down, may be to sub-bullets?

File: modules/op-constructing-pipelines-using-pipeline-builder.adoc [hunk 3]
```diff
@@ -30,13 +30,20 @@ image::op-pipeline-yaml.png[]
...
++
+[NOTE]
+====
+The search list contains all the Tekton Hub tasks and tasks available in the cluster. Also, if a task is already installed it will show *Add* to add the task whereas it will show *Install and add* to install and add the task. It will show *Update and add* when you add the same task with an updated version.
+====
+
+*** To add sequential tasks to the pipeline, click the plus icon to the right or left of the task, click *Add task*, search for a task in the quick search, select the required task from the list and click *Add* or *Install and add*.
```
Comment (sounix000) [inline modules/op-constructing-pipelines-using-pipeline-builder.adoc:40]: The sentence seems to be quite complex, in the sense that it contains several sequential actions. Can we break it down, may be to sub-bullets?

File: modules/op-constructing-pipelines-using-pipeline-builder.adoc [hunk 4]
```diff
@@ -30,13 +30,20 @@ image::op-pipeline-yaml.png[]
...
 +
 .Pipeline Builder
 image::op-pipeline-builder.png[]
-+
+
+*** To add a final task, click the *Add finally task* and click *Add task*, search for a task in the quick search, select the required task from the list and click *Add* or *Install and add*.
```
Comment (sounix000) [inline modules/op-constructing-pipelines-using-pipeline-builder.adoc:45]: The sentence seems to be quite complex, in the sense that it contains several sequential actions. Can we break it down, may be to sub-bullets? For example:
*** Add a final task:
**** Click *Add finally task* -> *Add task*
**** Search for a task in the quick search, and select the required task from the displayed list. 
**** Click *Add*, or *Install and add*.

**Pull request merged**